packages/ra-core/src/reducer/admin/resource/data.js

```diff
@@ -60,40 +57,43 @@ const addRecords = addRecordsFactory(getFetchedAt);
 const initialState = {};
 Object.defineProperty(initialState, 'fetchedAt', { value: {} }); // non enumerable by default
 
-export default (previousState = initialState, { type, payload, meta }) => {
-    if (type === CRUD_UPDATE_OPTIMISTIC) {
-        const updatedRecord = { ...previousState[payload.id], ...payload.data };
-        return addRecords([updatedRecord], previousState);
-    }
-    if (type === CRUD_UPDATE_MANY_OPTIMISTIC) {
-        const updatedRecords = payload.ids
-            .reduce((records, id) => records.concat(previousState[id]), [])
-            .map(record => ({ ...record, ...payload.data }));
-        return addRecords(updatedRecords, previousState);
-    }
-    if (type === CRUD_DELETE_OPTIMISTIC) {
-        const { [payload.id]: removed, ...newState } = previousState;
+export default (previousState = initialState, { payload, meta }) => {
+    if (meta && meta.optimistic) {
+        if (meta.fetch === UPDATE) {
```

Comment:
A switch would be better IMO

Reply:
I disagree, I won't have a default case

Reply:
default case represent the case where meta.fetch has an unexpected value, it should be handled and the user warned. IMO

Reply:
`meta.fetch` can be anything the user wants, provided that they deal with it in their `dataProvider`. This is not REST, but really a dictionary of verbs. We shouldn't limit what the user can do, or send a warning when an unknown `meta.fetch` is used.

Reply:
ok